### PR TITLE
Add RDS option group submodule

### PR DESCRIPTION
**Pull request by JonRoma**
@kwessel submitted code for an RDS option group submodule.

For now, we won't fold in all the other RDS-related modules into a single module supporting multiple submodules. However, we can start the transition by adding this code to the terraform-aws-rds repository using the
    
Terraform's [standard module structure](https://developer.hashicorp.com/terraform/language/modules/develop/structure).

I've moved this into a branch (not my fork).

### CODE REVIEW

File: modules/option-group/main.tf [hunk 1]
```diff
@@ -0,0 +1,50 @@
+locals {
+  og_name = format("%s-%s-%s", var.name_prefix, var.engine, replace(var.engine_version, ".", "-"))
```
Comment (JonRoma) [inline modules/option-group/main.tf:2]: I see a typo: `og_name` rather than `log_name`.

File: modules/option-group/main.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        content {
+          name  = option_settings.value.name
+          value = option_settings.value.value
+        }
+      }
+    }
+  }
+
+  dynamic "timeouts" {
```
Comment (JonRoma) [inline modules/option-group/main.tf:39]: This might be worth rethinking.

**Review by kwessel — APPROVED** [review]

File: modules/option-group/main.tf (hunk 1 above)
Comment (kwessel) [reply modules/option-group/main.tf:2]: No, it's supposed to be og_name. og meaning option group. No idea what a log_name would be in this context. I based that on pg_name from the parameter group module.

Comment (JonRoma) [reply modules/option-group/main.tf:2]: That makes sense now that you explain it. I was thinking that you were trying to compete with me for being error prone today. :-D

**Comment by JonRoma** [comment]
@kwessel, this has been merged and is tagged with `v3.2.1`.

**Pull request merged**